pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -60,87 +65,85 @@ func (current *Resources) Remove(incoming Resources, all bool) {
 // an aggregated error will be returned with all exceeded resources.
 func (current *Resources) Fits(incoming Resources) error {
 	exceededResources := []string{}
-	if current.Apps < incoming.Apps {
-		exceededResources = append(exceededResources, "Apps")
-	}
-	if current.Containers < incoming.Containers {
-		exceededResources = append(exceededResources, "Containers")
-	}
-	if current.Jobs < incoming.Jobs {
-		exceededResources = append(exceededResources, "Jobs")
-	}
-	if current.Volumes < incoming.Volumes {
-		exceededResources = append(exceededResources, "Volumes")
-	}
-	if current.Secrets < incoming.Secrets {
-		exceededResources = append(exceededResources, "Secrets")
-	}
-	if current.Images < incoming.Images {
-		exceededResources = append(exceededResources, "Images")
-	}
 
-	if current.VolumeStorage.Cmp(incoming.VolumeStorage) < 0 {
-		exceededResources = append(exceededResources, "VolumeStorage")
-	}
-	if current.Memory.Cmp(incoming.Memory) < 0 {
-		exceededResources = append(exceededResources, "Memory")
+	// Define function for checking int resources to keep code DRY
+	checkResource := func(resource string, currentVal, incomingVal int) {
+		if currentVal <= incomingVal {
+			exceededResources = append(exceededResources, resource)
+		}
 	}
-	if current.CPU.Cmp(incoming.CPU) < 0 {
-		exceededResources = append(exceededResources, "Cpu")
+
+	// Define function for checking quantity resources to keep code DRY
+	checkQuantityResource := func(resource string, currentVal, incomingVal resource.Quantity) {
+		if currentVal.Cmp(incomingVal) <= 0 {
+			exceededResources = append(exceededResources, resource)
+		}
 	}
 
+	checkResource("Apps", current.Apps, incoming.Apps)
+	checkResource("Containers", current.Containers, incoming.Containers)
+	checkResource("Jobs", current.Jobs, incoming.Jobs)
+	checkResource("Volumes", current.Volumes, incoming.Volumes)
+	checkResource("Secrets", current.Secrets, incoming.Secrets)
+	checkResource("Images", current.Images, incoming.Images)
+	checkResource("Projects", current.Projects, incoming.Projects)
+
+	checkQuantityResource("VolumeStorage", current.VolumeStorage, incoming.VolumeStorage)
+	checkQuantityResource("Memory", current.Memory, incoming.Memory)
+	checkQuantityResource("Cpu", current.CPU, incoming.CPU)
```

Comment:
Mentioned in slack -- here's an alternative (take it or leave it):

```go
func (current *Resources) Fits(incoming Resources) error {
    // ...
	for _, cmp := range []struct{
	    resource string
	    limit, incoming any
	}{
	  {"Apps", current.Apps, incoming.Apps},
	  // ...
	  {"Cpu", current.CPU, incoming.CPU},
	}{
	   if gt(cmp.incoming, cmp.limit) {
	       exceeded = append(exceeded, cmp.resource)
	   }
	}
	//...
}

// gt returns true IFF a is greater than b.
func [T any]gt(a, b T) bool {
   switch v := a.(type) {
   case *resource.Quantity:
     return v.Cmp(b.(resource.Quantity)) > 0
   case int:
     return v > b.(int)
   }

   return false
}
```

Reply:
you could also do something like `runtime.Scheme`, where -- using reflection -- you register comparator functions for different types, then you wouldn't need a type switch

anyway, this is a nice exercise, but I don't feel that any of these approaches are _more_ understandable or practical than the approach being replaced by this PR.

Reply:
This is super interesting. I do think that what you said is right in that it is cool but isn't moving the needle of understanding. If you're okay with, let's get the PR in as is?